src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -287,40 +291,7 @@ private void UpdateSolutionFile()
             foreach (ConfigFileInformation info in bindingConfigInformationMap.Values)
             {
                 Debug.Assert(fileSystem.File.Exists(info.NewFilePath), "File not written " + info.NewFilePath);
-                this.AddFileToSolutionItems(info.NewFilePath);
-                this.RemoveFileFromSolutionItems(info.NewFilePath);
-            }
-        }
-
-        private void AddFileToSolutionItems(string fullFilePath)
-        {
-            Debug.Assert(Path.IsPathRooted(fullFilePath) && fileSystem.File.Exists(fullFilePath), "Expecting a rooted path to existing file");
-
-            Project solutionItemsProject = this.projectSystem.GetSolutionFolderProject(Constants.LegacySonarQubeManagedFolderName, true);
-            if (solutionItemsProject == null)
-            {
-                Debug.Fail("Could not find the solution items project");
-            }
-            else
-            {
-                if (!this.projectSystem.IsFileInProject(solutionItemsProject, fullFilePath))
```

Comment:
Ok, so this is the duplicated code that has been removed.

There are slight differences between this version of the code and the version that replaces it i.e. this version calls `IsFileInProject`.
Going forwards, when sending out a review please
* call out the duplicated code that has been removed, and
* if there are differences between the old and new versions, call them out.
Otherwise, it's a real pain to find and review the de-duplicated code in the GitHub UI.

Reply:
IsFileInProject is done internally inside projectSystemHelper, that's why I went with the version that didn't re-check it.